### PR TITLE
feat: Add NATS and JetStream Prometheus alert rules

### PR DESCRIPTION
**Pull request by somratdutta**
- Added multiple Prometheus alert rules for monitoring NATS server and JetStream metrics.

- Included alerts for:
  - High connection count
  - High pending bytes
  - High subscriptions count
  - High routes count
  - High memory usage
  - Slow consumers
  - NATS server downtime
  - High CPU usage
  - High number of active connections
  - High JetStream store and memory usage
  - Subscription limits exceeded
  - High pending messages
  - Authentication timeouts
  - Errors in NATS (JetStream API errors)
  - JetStream consumers limit exceeded
  - Exceeding max payload size
  - Leaf node connection issues
  - Ping operations limit exceeded
  - Write deadline exceeded
- Ensured consistency between `exporter.yml` and `rules.yml` files.

This commit enhances the visibility of NATS and JetStream operations by providing key metrics to alert on potential issues and optimize system performance.

### CODE REVIEW

**Comment by somratdutta** [comment]
@samber 
there was a rising demand for important nats prometheus alerts in the community, so decided to contribute to it.
Let me know for any changes, would love to contribute to this.

File: _data/rules.yml [hunk 1]
```diff
@@ -1532,9 +1532,90 @@ groups:
...
+                severity: warning
+                for: 5m
+              - name: Nats high number of subscriptions
+                description: NATS server has more than 1000 active subscriptions
+                query: "gnatsd_connz_subscriptions > 1000"
+                severity: warning
+                for: 5m
+              - name: High pending messages in NATS
+                description: NATS server has more than 100,000 pending messages
+                query: "gnatsd_connz_pending_bytes > 100000"
```
Comment (samber) [inline _data/rules.yml:1580]: 100k pending messages or 100kB ?

File: _data/rules.yml [hunk 2]
```diff
@@ -1532,9 +1532,90 @@ groups:
...
+                severity: warning
+                for: 5m
+              - name: Nats slow consumers
+                description: There are slow consumers in NATS for {{ $labels.instance }}
+                query: "gnatsd_varz_slow_consumers > 0"
+                severity: critical
+                for: 3m
+              - name: Nats server down
+                description: NATS server has been down for more than 5 minutes
+                query: "absent(gnatsd_connz_total)"
```
Comment (samber) [inline _data/rules.yml:1550]: is it applicable when you have multiple NATS servers ?

File: _data/rules.yml [hunk 3]
```diff
@@ -1532,9 +1532,90 @@ groups:
                 for: 3m
               - name: Nats high routes count
                 description: High number of NATS routes ({{ $value }}) for {{ $labels.instance }}
-                query: "gnatsd_routez_num_routes > 10"
+                query: "gnatsd_varz_routes > 10"
                 severity: warning
                 for: 3m
+              - name: Nats high memory usage
+                description: NATS server memory usage is above 200MB for {{ $labels.instance }}
+                query: "gnatsd_varz_mem > 200000000"
```
Comment (samber) [inline _data/rules.yml:1540]:
```suggestion
                query: "gnatsd_varz_mem > 200 * 1024 * 1024"
```

(more readable)

**Review by samber — CHANGES_REQUESTED** [review]
hi @somratdutta and thanks for this first contribution \o/

**Comment by samber** [comment]
FYI, you don't need to update the dist/ files. It is generated on merge.

**Comment by somratdutta** [comment]
thanks @samber for the quick response, 
I would make the suggested changes and get back.
😄

File: _data/rules.yml (hunk 2 above)
Comment (somratdutta) [reply _data/rules.yml:1550]: i am using single server setup so I didn't think about this corner case.
We can use the up PromQL exprr instead then.
`up{job="nats"}`

File: _data/rules.yml (hunk 1 above)
Comment (somratdutta) [reply _data/rules.yml:1580]: 100kB,
sorry for the confusion

**Comment by somratdutta** [comment]
> FYI, you don't need to update the dist/ files. It is generated on merge.

sure, will revert the changes

File: _data/rules.yml (hunk 2 above)
Comment (samber) [reply _data/rules.yml:1550]: Yes, you can find some examples with other exporters

**Comment by somratdutta** [comment]
hi @samber can you review again?
i have worked on the comments 😄

**Review by samber — APPROVED** [review]
lets go!

**Pull request merged**